### PR TITLE
Feature/silence channels

### PR DESCRIPTION
**Pull request by hanaum**

### CODE REVIEW

File: js/src/http/UserDirectoryApi.coffee [hunk 1]
```diff
@@ -166,4 +168,33 @@ define 'kryptnostic.user-directory-api', [
         })
       ))
 
+    getMutedChannels: ->
+      Promise.resolve(axios(
+        Requests.wrapCredentials({
+          url: getUserUrl() + '/setting/' + MUTED_CHANNEL
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: this looks wrong. the URL will become `'/directory/user/setting/SILENCED_CHANNELS'`. are you sure this is the correct URL?

File: js/src/http/UserDirectoryApi.coffee [hunk 2]
```diff
@@ -166,4 +168,33 @@ define 'kryptnostic.user-directory-api', [
...
+          headers: DEFAULT_HEADERS
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: you don't need `headers` on `GET` requests

File: js/src/http/UserDirectoryApi.coffee [hunk 3]
```diff
@@ -26,7 +26,9 @@ define 'kryptnostic.user-directory-api', [
 
   log = Logger.get('UserDirectoryApi')
 
-  DEFAULT_HEADER = { 'Content-Type' : 'application/json' }
+  DEFAULT_HEADERS = { 'Content-Type' : 'application/json' }
+  MUTED_CHANNEL = 'SILENCED_CHANNELS'
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: the concept of "channel" should not exist in KJS. why are you introducing this here?

File: js/src/http/UserDirectoryApi.coffee (hunk 1 above)
Comment (Bloodyhacker) [inline js/src/http/UserDirectoryApi.coffee]: Yes, this is the updated URL

Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: then the value for setting should be passed in, not set inside KJS

Comment (hanaum) [inline js/src/http/UserDirectoryApi.coffee]: I see. Will fix.

Comment (hanaum) [inline js/src/http/UserDirectoryApi.coffee]: Fixed. Now passing in value from ocelot.

File: js/src/http/UserDirectoryApi.coffee (hunk 2 above)
Comment (hanaum) [inline js/src/http/UserDirectoryApi.coffee]: Fixed.

File: js/src/http/UserDirectoryApi.coffee [hunk 4]
```diff
@@ -166,4 +166,32 @@ define 'kryptnostic.user-directory-api', [
         })
       ))
 
+    getMutedChannels: (setting) ->
+      Promise.resolve(axios(
+        Requests.wrapCredentials({
+          url: getUserUrl() + '/setting/' + setting
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: create a helper function for this URL:

    getUserSettingUrl = (userSetting) -> getUserUrl() + '/setting/' + userSetting

Comment (hanaum) [inline js/src/http/UserDirectoryApi.coffee]: Makes sense. Fixed.

File: js/src/http/UserDirectoryApi.coffee [hunk 5]
```diff
@@ -166,4 +167,32 @@ define 'kryptnostic.user-directory-api', [
         })
       ))
 
+    getMutedChannels: (setting) ->
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: the concept of "channels" doesn't exist in KJS. it seems like this is actually just getting the user setting, so perhaps rename the function to `getUserSetting`

File: js/src/http/UserDirectoryApi.coffee [hunk 6]
```diff
@@ -166,4 +167,32 @@ define 'kryptnostic.user-directory-api', [
...
+      Promise.resolve(axios(
+        Requests.wrapCredentials({
+          url: getUserSettingUrl(setting)
+          method: 'GET'
+        })
+      ))
+
+    addMutedChannel: (uuid, setting) ->
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: the concept of "channels" doesn't exist in KJS. it seems like this is just adding a user setting, so perhaps rename the function to `setUserSetting` or `addUserSetting`

File: js/src/http/UserDirectoryApi.coffee [hunk 7]
```diff
@@ -166,4 +167,32 @@ define 'kryptnostic.user-directory-api', [
...
+      Promise.resolve(axios(
+        Requests.wrapCredentials({
+          url: getUserSettingUrl(setting)
+          headers: DEFAULT_HEADERS
+          method: 'PUT'
+          data: [uuid]
+        })
+      ))
+
+    removeMutedChannel: (uuid, setting) ->
```
Comment (UnsungHero97) [inline js/src/http/UserDirectoryApi.coffee]: the concept of "channels" doesn't exist in KJS. it seems like this is just deleting a user setting, so perhaps rename the function to `removeUserSetting`

Comment (Bloodyhacker) [inline js/src/http/UserDirectoryApi.coffee]: +1

**Pull request merged**